From c77b8f23af7d285fd82dbd3e4db3fa2c09899e58 Mon Sep 17 00:00:00 2001
From: James Le Cuirot <chewi@gentoo.org>
Date: Fri, 9 Mar 2018 22:39:41 +0000
Subject: [PATCH 1/2] build: Fix configure --without-ipv6-default behaviour

The current behaviour disables ipv6-default when no switch is given at
all but otherwise checks if libtirpc was requested, regardless of
whether you have given --with-ipv6-default or --without-ipv6-default.

I believe the intention was to enable when libtirpc is requested by
default but otherwise respect the switch given.

This is important because ipv6-default breaks Gluster for systems that
have IPv6 disabled.

BUG: 1553926
Change-Id: I76b91ae2699574b2e5b777453732bb5cbd79bbca
Signed-off-by: James Le Cuirot <chewi@gentoo.org>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 827ec0518..d42179182 100644
--- a/configure.ac
+++ b/configure.ac
@@ -315,7 +315,7 @@ AC_ARG_WITH([libtirpc],
 
 AC_ARG_WITH([ipv6-default],
         AC_HELP_STRING([--with-ipv6-default], [Set IPv6 as default.]),
-        [with_ipv6_default=$with_libtirpc], [with_ipv6_default=no])
+        [], [with_ipv6_default=$with_libtirpc])
 
 if test "x$ac_cv_file__etc_redhat_release" = "xyes"; then
         if rpm -qa centos-release | grep centos; then
-- 
2.16.1

